modules/internal/ChapelBase.chpl

```diff
@@ -441,7 +441,7 @@ module ChapelBase {
   //
   // bitwise operations on primitive types
   //
-  inline proc ~(a: bool) return __primitive("u~", a);
+  inline proc ~(a: bool) { compilerError("~ is not suported on bool"); }
```

Comment:
I think that this should have the "compiler generated" ("last resort") pragma applied to it so that a user or package module could choose to provide its own overload if it didn't like our choice.

In other similar error messages, we tend to use phrases like "...on operands of type bool" or "...on operands of boolean type" which I find more intuitive than simply "...on bool" (which, taken literally, suggests to me that perhaps the operation is not supported on the ``bool`` type itself?).  If keeping the message succinct seems important, even saying "on bools" or "on boolean operands" seems clearer.